tinkar-maven-plugin/src/main/java/dev/ikm/maven/export/VerifyPBExport.java

```diff
@@ -0,0 +1,257 @@
+import dev.ikm.tinkar.common.id.IntIdSet;
+import dev.ikm.tinkar.common.id.PublicId;
+import dev.ikm.tinkar.entity.ConceptEntity;
+import dev.ikm.tinkar.entity.ConceptEntityVersion;
+import dev.ikm.tinkar.entity.Entity;
+import dev.ikm.tinkar.entity.EntityService;
+import dev.ikm.tinkar.entity.PatternEntity;
+import dev.ikm.tinkar.entity.PatternEntityVersion;
+import dev.ikm.tinkar.entity.SemanticEntity;
+import dev.ikm.tinkar.entity.SemanticEntityVersion;
+import dev.ikm.tinkar.entity.StampEntity;
+import dev.ikm.tinkar.entity.StampEntityVersion;
+import dev.ikm.tinkar.entity.load.LoadEntitiesFromProtobufFile;
+import dev.ikm.tinkar.integration.DataIntegrity;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.shared.model.fileset.FileSet;
+import org.apache.maven.shared.model.fileset.util.FileSetManager;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Mojo(name = "verify-export", requiresDependencyResolution = ResolutionScope.RUNTIME_PLUS_SYSTEM, defaultPhase = LifecyclePhase.VERIFY)
+public class VerifyPBExport extends TinkarMojo {
+
+	private final FileSetManager fileSetManager = new FileSetManager();
+
+	@Parameter(name= "filesets")
+	private FileSet[] filesets;
+
+	@Parameter(name = "fileset")
+	private FileSet fileset;
+
+	@Parameter(defaultValue = "${project}")
+	private MavenProject project;
+
+	@Isolate
+	private List<File> filesToLoad = new ArrayList<>();
+
+	@Override
+	public void handleIsolatedFields() {
+		if (filesets != null) {
+			for (FileSet innerFileSet : filesets) {
+				for (String includeFile : fileSetManager.getIncludedFiles(innerFileSet)) {
+					File pbZip = new File(innerFileSet.getDirectory(), includeFile);
+					filesToLoad.add(pbZip);
+				}
+			}
+		}
+		if (fileset != null) {
+			for (String includeFile : fileSetManager.getIncludedFiles(fileset)) {
+				File pbZip = new File(fileset.getDirectory(), includeFile);
+				filesToLoad.add(pbZip);
+			}
+		}
+	}
+
+	@Override
+	public void run() {
+		if (!isolate) {
+			handleIsolatedFields();
+		}
+		filesToLoad.forEach(file -> {
+			var loadTask = new LoadEntitiesFromProtobufFile(file);
+		});
+
+		StringBuilder sb = new StringBuilder();
+
+		Map<String, List<? extends Entity>> typeNameEntityMap = new HashMap<>();
+		List<Integer> aggregatedNullNidList = new ArrayList<>();
+		boolean failed = false;
+
+		List<? extends Entity> entities = DataIntegrity.validateStampReferences(aggregatedNullNidList);
+		typeNameEntityMap.put("Stamp", entities);
+
+		if (!entities.isEmpty()) {
```

Comment:
Maybe move each check into separate methods, instead of placing everything under `run()`

Reply:
Fixed in latest commit